packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -1014,6 +1026,81 @@ export const useArbTokenBridge = (
     )
   }
 
+  // mutates provided array - assigns deadlineBlockNumbers to all asserted but unconfimred l2tol1 messages 
+  const addNodeDeadlineToUnconfirmedWithdrawals = async(l2ToL1Data: L2ToL1EventResultPlus[])=>{
+    if(l2ToL1Data.length === 0) return []
+    const l1NetworkID = await l1NetworkIDCached()
+    const oldOutboxOffset = (()=>{
+      switch (l1NetworkID) {
+        case '1':
+          return 30
+        case '4':
+          return 326
+        default:
+          throw new Error(`Unrecognized L1 network ${l1NetworkID}`)
+      }
+    })()
+
+    // get smallest batch number in messages for lower bound on graph query    
+    const smallestBatchNumber = l2ToL1Data.reduce((acc: BigNumber, currentL2ToL1Data:L2ToL1EventResultPlus )=>{
```

Comment:
could be easier/cleaner to sort then get the first value

Reply:
If we sort `L2ToL1EventResultPlus` when we first construct it, could maybe be used in other places too

Reply:
Ah we already sort it a couple lines down 👀 
```// ensure sorted in ascending order by timestamp```
can use that here, if its sorted by timestamp it also is by batchNumber since its only increasing